blockchain/accept.go

```diff
@@ -52,7 +52,7 @@ func (b *BlockChain) maybeAcceptBlock(block *btcutil.Block, flags BehaviorFlags)
 	// such as making blocks that never become part of the main chain or
 	// blocks that fail to connect available for further analysis.
 	err = b.db.Update(func(dbTx database.Tx) error {
-		return dbMaybeStoreBlock(dbTx, block)
+		return dbStoreBlock(dbTx, block)
```

Comment:
The assumption is that `maybeAcceptBlock` is not called if the block is already stored. Which is true: ProcessBlock checks the block index before calling `maybeAcceptBlock` and `processOrphans` only will call it will orphan blocks that couldn't previously be connected.

Reply:
Fair point.  With this change to put all blocks into the block index, that check will always catch it.  That wasn't the case before there when the memory block index only contained the (part of) the main chain and then later the entire main chain.

Once there is full header sync, all of the orphan processing in `blockchain` should be able to go away completely since it will only ever be processing blocks that have already been proven to connect via the headers.